.github/workflows/pull-request-ci.yaml

```diff
@@ -199,14 +199,22 @@ jobs:
           echo "::set-env name=SSH_AUTH_SOCK::${SSH_AUTH_SOCK}"
           echo "::set-env name=SSH_AGENT_PID::${SSH_AGENT_PID}"
 
-      # set GKE secret. TBD: use vault instead
+      - name: Query vault for GKE secrets
+        id: vault
+        uses: hashicorp/vault-action@v2.0.1
+        with:
+          exportEnv: false
+          url: https://volt.cap.explore.suse.dev/
+          token: ${{ secrets.VAULT_TOKEN }}
+          secrets: |
+            secret/data/cloud-providers gke.GKE_CRED_JSON | GKE_CRED_JSON
+
+      # set GKE secret.
       - name: set GKE_CRED_JSON
         run: |
           json_file="$(mktemp)"
-          echo "$GKE_CRED_JSON" > "${json_file}"
+          echo "${{ steps.vault.outputs.GKE_CRED_JSON }}" > "${json_file}"
```

Comment:
Hmm. Technically this means the token will be in the process command line here, right? (Though I guess `echo` is _probably_ a built in and doesn't have a real command line?)  Do we have to worry about it showing up in the bash history? (This probably isn't interactive, so… probably not?)

Generally using the environment is better.

Reply:
I agree that `echo` without `set -x` is ok, is a built-in of the shell.

I wonder how GH Actions redacts things, e.g:
https://github.com/cloudfoundry-incubator/kubecf/runs/1216923853?check_suite_focus=true#step:10:3.
Looking at https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idoutputs, it seems that the secret `steps.vault.outputs.GKE_CRED_JSON` would get redacted in the worker and never shared with GHA.

> Generally using the environment is better.

I don't personally agree, as env can get leaked to subprocesses and utilities you call (as normally you don't unset them), and env usually gets logged or dumped somewhere. I try to always use files, in memory with tmpfs, mktemp (what we are doing here), named fifos, or by mounting the files into whatever corresponds. If you were to have access to the file, you would have access to the env anyways. But at least you don't usually log the file contents somewhere.

Reply:
PR is rebased. Left this as-is for now.